core/src/main/java/io/grpc/internal/RetriableStream.java

```diff
@@ -817,6 +817,7 @@ private final class Sublistener implements ClientStreamListener {
 
     @Override
     public void headersRead(final Metadata headers) {
+      headers.put(GRPC_PREVIOUS_RPC_ATTEMPTS, String.valueOf(substream.previousAttemptCount));
```

Comment:
Do you want to do this conditionally only if `substream.previousAttemptCount > 0` ?